src/fmu/dataio/_model/global_configuration.py

```diff
@@ -137,6 +137,36 @@ def __iter__(self) -> Any:
     def __getitem__(self, item: str) -> StratigraphyElement:
         return self.root[item]
 
+    def _get_index(self, name: str) -> int:
+        """
+        Get the index of a stratigraphic element by name.
+        """
+        return list(self.root.keys()).index(name)
+
+    def get_official_name(self, name: str) -> str:
+        """
+        Get the official SMDA name of a stratigraphic element.
+        name: name of stratigraphic element
+        """
+        if name in self.root:
+            return self.root[name].name
+        return ""
+
+    def get_ordered_official_names(self, names: list[str]) -> list[str]:
+        """
+        Retrieve the official SMDA names and order them according to
+        the stratigraphic column.
+        names: names of stratigraphic elements
+        """
+        try:
+            sorted_indices = sorted([self._get_index(name) for name in names])
```

Comment:
Can you comment on this method of getting the sorted indices, over something like `enumerate`? I am guessing it's to handle the case where some indices may be skipped, like `[1, 2, 5, 7, 8]`?

Reply:
Yes, exactly.
The reason being that the formations listed in the juxtapositions may be a subset of the formations in the stratigraphic column. This is a general approach that also covers the case when there are gaps in the formations listed in the juxtapositions compared to the formations in the stratigraphic column, e.g. [2, 3, 5].